packages/interactivity/src/utils.ts

```diff
@@ -48,12 +56,14 @@ const afterNextFrame = ( callback: () => void ) => {
  *
  * @return Promise
  */
-export const splitTask = () => {
-	return new Promise( ( resolve ) => {
-		// TODO: Use scheduler.yield() when available.
-		setTimeout( resolve, 0 );
-	} );
-};
+export const splitTask =
+	typeof window.scheduler?.yield === 'function'
+		? window.scheduler.yield.bind( window.scheduler )
```

Comment:
In other places in Gutenberg, I believe we use `requestIdleCallback` to optimize for UI responsiveness and delay some tasks with lower priority. I wonder if you can speak about the difference and when to favor one over the other.

Reply:
With `requestIdleCallback` the idleness may occur several seconds after requested which would not be desirable for hydration. For `splitTask` it's purely to break up long tasks which remain high priority. So both have their use cases.